qiime2/sdk/tests/test_usage.py

```diff
@@ -0,0 +1,259 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+import tempfile
+
+from qiime2.core.type import signature
+from qiime2.core.testing.util import get_dummy_plugin
+from qiime2.core.testing.type import Mapping
+from qiime2.sdk import usage, action
+from qiime2 import plugin
+
+
+class TestCaseUsage(unittest.TestCase):
+    def setUp(self):
+        self.test_dir = tempfile.TemporaryDirectory(prefix='qiime2-test-temp-')
+        self.plugin = get_dummy_plugin()
+
+    def tearDown(self):
+        self.test_dir.cleanup()
+
+
+class TestUsage(TestCaseUsage):
+    def test_basic(self):
+        action = self.plugin.actions['concatenate_ints']
+        use = usage.DiagnosticUsage()
```

Comment:
It's like a mock, but not because it's just as valid an implementation as anything else! :zap: